### DISCUSSION TITLE
Micro-proposal: Implicit Serialize/Deserialize implementations

### DISCUSSION THREAD

**Discussion by cmxog** [tact-lang/tact-bikeshed#4] [Ideas]
Seeing that [de]serialization is a very common problem in smart contracts and observing the code polluted with the following snippet:

```
@derive impl Serialize {}
@derive impl Deserialize {}
```

I propose to implicitly implement these interfaces for types that are being serialized (directly or indirectly) and fail compilation if it's impossible (say, `Integer` is used instead of `IntXX`)

This will reduce the length of contracts, decrease the cognitive load, and still allow for explicit custom implementations where necessary.

**Comment by oleganza** [comment]
Yeah, i suggest we simply run with "one serialization rules them all" with the rule "Tact types are mappable 1:1 to TL-B".
